### PR TITLE
The center of drop and jump patterns is shifted from real value #291

### LINKED ISSUE #291: The center of  drop and jump patterns is shifted from real value
We need to change the work of methods: they were better looking for jump or drops centers .
Problem is that `segment_cent_index` in jump and drop models is shifted from real pattern center for a few values.
Possible workaround: remove smoothed data in methods, because in a small window size (for example 20 values) smoothing noticeably changes data..

### PR DESCRIPTION
**Pull request by VargBurz**
Closes #291

### CODE REVIEW

File: analytics/analytics/utils/common.py [hunk 1]
```diff
@@ -371,7 +370,7 @@ def get_distribution_density(segment: pd.Series) -> float:
         segment_max_line = ax_list[max_peak_index, 0]
         segment_median = ax_list[antipeaks_kde[0], 0]
     except IndexError:
-        segment_max_line = max_jump
-        segment_min_line = min_jump
+        segment_max_line = max_jump * 0.95
```
Comment (amper43) [inline analytics/analytics/utils/common.py]: please, add const or comment that eplains this magic constants

File: analytics/analytics/utils/common.py [hunk 2]
```diff
@@ -182,17 +182,16 @@ def find_drop_length(segment_data, min_line, max_line):
...
-    for i in range(len(segment_data)):
-        f.append(median_line)
-    f = np.array(f)
-    g = []
-    for i in segment_data:
-        g.append(i)
-    g = np.array(g)
-    idx = np.argwhere(np.diff(np.sign(f - g)) != 0).reshape(-1) + 0
-    return idx
+    cen_ind = []
```
Comment (amper43) [inline analytics/analytics/utils/common.py]: it is too complex for understanding, can you add simple docstring that explain which data will be deleted?

File: analytics/analytics/utils/common.py [hunk 3]
```diff
@@ -7,6 +7,10 @@
 from typing import Union
 import utils
 
+SHIFT_FACTOR = 0.05
+
```
Comment (jonyrock) [inline analytics/analytics/utils/common.py]:
```suggestion
```

File: analytics/analytics/utils/common.py [hunk 4]
```diff
@@ -64,7 +68,7 @@ def segments_box(segments):
 
 def intersection_segment(data, median):
     """
-    Finds all intersections between flatten data and median
+    Finds all intersections between drop pattern data and median
     """
     cen_ind = []
     for i in range(1, len(data)-1):
```
Comment (jonyrock) [inline analytics/analytics/utils/common.py]:
```suggestion
    for i in range(1, len(data) - 1):
```

File: analytics/analytics/utils/common.py [hunk 5]
```diff
@@ -64,7 +68,7 @@ def segments_box(segments):
 
 def intersection_segment(data, median):
```
Comment (jonyrock) [inline analytics/analytics/utils/common.py]: the name should include a verb. Now it looks like a name for a class.

Comment (jonyrock) [inline analytics/analytics/utils/common.py]:
```suggestion
def find_intersection_segments(data, median):
```

Comment (jonyrock) [inline analytics/analytics/utils/common.py]: add return type

File: analytics/analytics/utils/common.py [hunk 6]
```diff
@@ -182,17 +186,22 @@ def find_drop_length(segment_data, min_line, max_line):
...
-        f.append(median_line)
-    f = np.array(f)
-    g = []
-    for i in segment_data:
-        g.append(i)
-    g = np.array(g)
-    idx = np.argwhere(np.diff(np.sign(f - g)) != 0).reshape(-1) + 0
-    return idx
+    """
+    Finds all intersections between flatten data and median
```
Comment (jonyrock) [inline analytics/analytics/utils/common.py]:
```suggestion
        Finds all intersections between flatten data and median
```

File: analytics/analytics/utils/common.py [hunk 7]
```diff
@@ -182,17 +186,22 @@ def find_drop_length(segment_data, min_line, max_line):
...
+    """
+    cen_ind = []
+    for i in range(1, len(segment_data)-1):
+        if segment_data[i - 1] > median_line and segment_data[i + 1] < median_line:
+            cen_ind.append(i)
+    """
+    Delete close values except the last one
```
Comment (jonyrock) [inline analytics/analytics/utils/common.py]:
```suggestion
        Delete close values except the last one
```

File: analytics/analytics/utils/common.py [hunk 8]
```diff
@@ -334,23 +343,23 @@ def find_jump_parameters(segment_data: pd.Series, segment_from_index: int):
...
     jump_length = utils.find_jump_length(segment_data, segment_min_line, segment_max_line) # finds all interseprions with median
-    cen_ind = utils.intersection_segment(flat_segment.tolist(), segment_median)
+    cen_ind = utils.intersection_segment(segment_data.tolist(), segment_median)
     jump_center = cen_ind[0]
-    segment_cent_index = jump_center - 5 + segment_from_index
+    segment_cent_index = jump_center + segment_from_index
     return segment_cent_index, jump_height, jump_length
 
 
 def find_drop_parameters(segment_data: pd.Series, segment_from_index: int):
```
Comment (jonyrock) [inline analytics/analytics/utils/common.py]: please remove a new line before this line

File: analytics/analytics/utils/common.py [hunk 9]
```diff
@@ -62,12 +65,12 @@ def segments_box(segments):
     max_time = pd.to_datetime(max_time, unit='ms')
     return min_time, max_time
 
-def intersection_segment(data, median):
+def find_intersection(data: pd.Series, median: float) -> list:
```
Comment (jonyrock) [inline analytics/analytics/utils/common.py]:
```suggestion
def find_intersections(data: pd.Series, median: float) -> list:
```

File: analytics/analytics/utils/common.py [hunk 10]
```diff
@@ -181,18 +184,21 @@ def find_drop_length(segment_data, min_line, max_line):
...
-    for i in range(len(segment_data)):
-        f.append(median_line)
-    f = np.array(f)
-    g = []
-    for i in segment_data:
-        g.append(i)
-    g = np.array(g)
-    idx = np.argwhere(np.diff(np.sign(f - g)) != 0).reshape(-1) + 0
-    return idx
+def find_drop_intersection(segment_data: pd.Series, median_line: float) -> list:
```
Comment (jonyrock) [inline analytics/analytics/utils/common.py]:
```suggestion
def find_drop_intersections(segment_data: pd.Series, median_line: float) -> list:
```

File: analytics/analytics/utils/common.py [hunk 11]
```diff
@@ -181,18 +184,21 @@ def find_drop_length(segment_data, min_line, max_line):
...
+    """
+        Finds all intersections between flatten data and median
+    """
+    cen_ind = []
+    for i in range(1, len(segment_data)-1):
+        if segment_data[i - 1] > median_line and segment_data[i + 1] < median_line:
+            cen_ind.append(i)
+    #Delete close values except the last one
```
Comment (jonyrock) [inline analytics/analytics/utils/common.py]:
```suggestion
    # Delete close values except the last one
```

**Pull request merged**